tests/unit/frameworks/test_any_agent.py

```diff
@@ -1,18 +1,147 @@
+    AgentFramework.AGNO: "litellm.acompletion",
+    AgentFramework.OPENAI: "litellm.acompletion",
+    AgentFramework.SMOLAGENTS: "litellm.completion",
+    AgentFramework.LLAMA_INDEX: "litellm.acompletion",
+}
+
+
+def create_agent_with_model_args(framework: AgentFramework) -> AnyAgent:
+    """Helper function to create an agent with test model arguments"""
+    return AnyAgent.create(
+        framework,
+        AgentConfig(
+            model_id="gpt-4o",
+            model_args={
+                "temperature": TEST_TEMPERATURE,
+                "frequency_penalty": TEST_PENALTY,
+            },
+        ),
+    )
+
+
+async def mock_streaming_response(
+    *args: Any, **kwargs: Any
+) -> AsyncGenerator[dict[str, Any], None]:
+    """
+    Mock the streaming response from litellm's acompletion function.
+    Accepts all arguments that would be passed to the real acompletion function.
+    """
+    # First chunk with role
+    yield {
+        "choices": [
+            {
+                "delta": {"role": "assistant", "content": "The state "},
+                "index": 0,
+                "finish_reason": None,
+            }
+        ]
+    }
+
+    # Middle chunks with content
+    yield {
+        "choices": [
+            {"delta": {"content": "capital of "}, "index": 0, "finish_reason": None}
+        ]
+    }
+
+    yield {
+        "choices": [
+            {
+                "delta": {"content": "Pennsylvania is "},
+                "index": 0,
+                "finish_reason": None,
+            }
+        ]
+    }
+
+    # Final chunk with finish reason
+    yield {
+        "choices": [
+            {"delta": {"content": "Harrisburg."}, "index": 0, "finish_reason": "stop"}
+        ]
+    }
+
+
+class TestAgentCreation:
+    def test_create_any_with_framework(self, agent_framework: AgentFramework) -> None:
+        agent = AnyAgent.create(agent_framework, AgentConfig(model_id="gpt-4o"))
+        assert agent
+
+    def test_create_any_with_valid_string(
+        self, agent_framework: AgentFramework
+    ) -> None:
+        agent = AnyAgent.create(agent_framework.name, AgentConfig(model_id="gpt-4o"))
+        assert agent
+
+    def test_create_any_with_invalid_string(self) -> None:
+        with pytest.raises(ValueError, match="Unsupported agent framework"):
+            AnyAgent.create("non-existing", AgentConfig(model_id="gpt-4o"))
+
+
+# Tests for model arguments
+class TestModelArguments:
+    def test_model_args(
+        self, agent_framework: AgentFramework, mock_litellm_response: Any
+    ) -> None:
+        if agent_framework == AgentFramework.LLAMA_INDEX:
+            pytest.skip("LlamaIndex agent uses a litellm streaming syntax")
+
+        agent = create_agent_with_model_args(agent_framework)
+
+        # Patch the appropriate litellm import path for this framework
+        import_path = LITELLM_IMPORT_PATHS[agent_framework]
+        with patch(import_path) as mock_litellm:
```

Comment:
IIRC you could use this marginally more compact alternative?

```suggestion
        with patch(import_path, return_value=mock_litellm_response) as mock_litellm:
```